extensions/scarb-doc/src/db.rs

```diff
@@ -0,0 +1,120 @@
+use cairo_lang_compiler::project::{
+    update_crate_root, update_crate_roots_from_project_config, ProjectConfig,
+};
+use cairo_lang_defs::db::{DefsDatabase, DefsGroup};
+use cairo_lang_doc::db::{DocDatabase, DocGroup};
+use cairo_lang_filesystem::db::{
+    init_files_group, AsFilesGroupMut, FilesDatabase, FilesGroup, CORELIB_CRATE_NAME,
+};
+use cairo_lang_parser::db::{ParserDatabase, ParserGroup};
+use cairo_lang_semantic::db::{SemanticDatabase, SemanticGroup};
+use cairo_lang_semantic::inline_macros::get_default_plugin_suite;
+use cairo_lang_semantic::plugin::PluginSuite;
+use cairo_lang_starknet::starknet_plugin_suite;
+use cairo_lang_syntax::node::db::{SyntaxDatabase, SyntaxGroup};
+use cairo_lang_utils::Upcast;
+
+use salsa;
+
+/// The Cairo compiler Salsa database tailored for scarb-doc usage.
+#[salsa::database(
+    FilesDatabase,
+    ParserDatabase,
+    SyntaxDatabase,
+    DefsDatabase,
+    SemanticDatabase,
+    DocDatabase
+)]
+pub struct ScarbDocDatabase {
```

Comment:
FYI you can do the following if you wish (no need to change)

```suggestion
/// The Cairo compiler Salsa database tailored for scarb-doc usage.
#[salsa::database(
    FilesDatabase,
    ParserDatabase,
    SyntaxDatabase,
    DefsDatabase,
    SemanticDatabase,
    cairo_lang_doc::db::DocDatabase
)]
pub struct DocDatabase {
```